usaspending_api/common/query_with_filters.py

```diff
@@ -433,30 +435,43 @@ def generate_elasticsearch_query(cls, filter_values: List[str], query_type: _Que
         def_codes_query = []
         def_code_field = f"{nested_path}{'.' if nested_path else ''}disaster_emergency_fund_code{'s' if query_type != _QueryType.ACCOUNTS else ''}"
 
-        all_covid_defc = set(
-            DisasterEmergencyFundCode.objects.filter(group_name="covid_19").values_list("code", flat=True)
+        all_covid_iija_defc = set(
+            DisasterEmergencyFundCode.objects.filter(group_name__in=["covid_19", "infrastructure"]).values_list(
+                "code", "earliest_public_law_enactment_date"
+            )
         )
-        covid_filters = list(set(filter_values) & all_covid_defc)
-        other_filters = list(set(filter_values) - all_covid_defc)
+        all_covid_iija_defc = dict(all_covid_iija_defc)
 
-        covid_queries = [ES_Q("match", **{def_code_field: v}) for v in covid_filters]
+        other_filters = list(set(filter_values) - set(all_covid_iija_defc.keys()))
         other_queries = [ES_Q("match", **{def_code_field: v}) for v in other_filters]
 
-        if covid_queries:
+        covid_iija_filters = {k: all_covid_iija_defc[k] for k in filter_values if k in all_covid_iija_defc.keys()}
+        covid_iija_queries = []
+
+        # Filter on the `disaster_emergency_fund_code` AND `action_date` values for transactions
+        for code, enactment_date in covid_iija_filters.items():
+            if query_type == _QueryType.TRANSACTIONS:
```

Comment:
Great catch creating separate logic for Transactions when building up the `covid_iija_queries` list